src/service/chat.js

```diff
@@ -206,6 +206,58 @@ const moveChatToAnon = async ({
   }
 };
 
+const initChatFromPost = async ({source, id}) => {
+  let payload = {source};
+  switch (source) {
+    case 'post':
+      payload = {...payload, postId: id};
+      break;
+    case 'comment':
+      payload = {...payload, commentId: id};
+      break;
+    default:
+      break;
+  }
+
+  try {
+    const response = await api.post('/chat/init-chat-from-post', payload);
+    if (response?.status === 200) {
+      return Promise.resolve(response.data);
+    }
+
+    return Promise.reject(response.data);
+  } catch (e) {
+    if (e?.response?.data?.message) return Promise.reject(e?.response?.data?.message);
+    return Promise.reject(e);
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `initChatFromPost` seems to be well implemented. It takes an object with `source` and `id` properties, constructs a payload based on the source type, sends a POST request to the server, and handles the response properly. However, it's important to note that if the `source` is neither 'post' nor 'comment', the payload will only contain the `source` property. If this is not the intended behavior, you might want to add error handling for unsupported sources.

```diff
+    default:
+      throw new Error(`Unsupported source: ${source}`);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->